test/Microsoft.ML.OnnxTransformTest/OnnxTransformTests.cs

```diff
@@ -57,7 +57,7 @@ public OnnxTransformTests(ITestOutputHelper output) : base(output)
         {
         }
 
-        [Fact]
+        [ConditionalFact(typeof(Environment), nameof(Environment.Is64BitProcess))] // x86 fails with "An attempt was made to load a program with an incorrect format."
```

Comment:
I assume Onnx support (i.e. Microsoft.ML.Scoring) is 64-bit only.  @shmoradims @jignparm - can you confirm?

Reply:
Was there a response to this feedback? I think this seems like a good idea.  Or perhaps even deriving a few fact attributes like `OnnxFact` or `TensorflowFact`, etc.

---
In reply to: [222758463](https://github.com/dotnet/machinelearning/pull/1008#discussion_r222758463) [](ancestors = 222758463)

Reply:
I'm not sure its worth the added overhead to declare a new fact, which itself calls `Environment.Is64BitProcess`.

The current conditional fact requires no "new" code and clearly defines that it is only supported on 64-bit processes.

Reply:
And when we try to add `Aarch64` (i.e. `arm64`)?

Reply:
Then we will need to re-review any of the currently disabled tests and possibly change this based on whether they support ARM and/or ARM64.

Reply:
There are multiple advantages to taking the above approach:

1. It is self-documenting, like @danmosemsft said.
2. It is centralized, so if it ever needs to be updated, only need to update a single spot.
3. We can add any rules we want into the method, where you can't do this in a single attribute.

The only disadvantage I see is the small amount of effort it is going to take to make this change.

Reply:
@eerhardt, do you have any problem with logging a tracking issue to improve this?

I'd like to get this merged in order to unblock people and I don't think it's going to be as trivial as just defining some new attributes, since we will need to do something different for .NET Core (where we need to instead check the `S.R.InteropServices.Architecture` of the process)

Reply:
that's fine with me.